c7n/resources/rds.py

```diff
@@ -1241,3 +1241,125 @@ class resource_type(object):
+                resource: rds
+                filters:
+                  - type: db-parameter
+                    key: someparam
+                    op: eq
+                    value: someval
+
+    based on the `describe_db_instances API`_
+
+    .. _describe_db_instances API: http://boto3.readthedocs.io/en/latest/\
+    reference/services/rds.html#RDS.Client.describe_db_instances
+    """
+
+    schema = {
+        'type': 'object',
+        # Doesn't mix well with inherits that extend
+        'additionalProperties': False,
+        'required': ['type', 'key', 'op', ],  # omit value for future "isnull"
+        'properties': {
+            'type': {'enum': ['db-parameter']},
+            'key': {'type': 'string'},
+            'default': {'oneOf': [
+                {'type': 'string'},
+                {'type': 'boolean'},
+                {'type': 'number'}
+            ]},
+            'value': {'oneOf': [
+                {'type': 'string'},
+                {'type': 'boolean'},
+                {'type': 'number'}
+            ]},
+            'op': {'enum': OPERATORS.keys()}}}
+
+    permissions = ('rds:DescribeDBInstances', 'rds:DescribeDBParameters', )
+
+    def process(self, resources, event=None):
+        pkey = self.data.get('key', '')
+        op = OPERATORS.get(self.data.get('op', 'eq'))
+        target_value = self.data.get('value', '0')
+        default = self.data.get('default', '0')
+        results = []
+
+        client = local_session(self.manager.session_factory).client('rds')
+        for resource in resources:
+            for pg in resource['DBParameterGroups']:
+                group_name = pg[u'DBParameterGroupName']
+
+                # get the params for that group
+                if 'Parameters' in pg:  # pg.has_key('Parameters'):
+                    # use previously augmented data
+                    db_params = pg['Parameters']
+                else:
+                    db_params = client.describe_db_parameters(
+                        DBParameterGroupName=group_name)
+                    # save it for the next filter
+                    pg['Parameters'] = db_params
+
+                param_struct = None
+                for ps in db_params['Parameters']:
+                    if ps['ParameterName'] == pkey:
+                        param_struct = ps
+                        break
+                if param_struct:
+                    # just because we found a name, doesnt mean a value will
+                    # be present, so use the default here too
+                    param_value = param_struct.get('ParameterValue', default)
+                else:
+                    param_value = default
+
+                # at this point the types of param_value and target_value may
+                # not match, so we let Amazon tell us how to interpret the
+                # values
+                dt = param_struct['DataType']
+
+                def recast(val, datatype):
```

Comment:
make this a static method on the class, inlining it in the middle of flow makes both harder to read. also ideally separate tests. there's no closure values extant.